### PR TITLE
Add FP8 Support

### PR DESCRIPTION
**Pull request by Satrat**
Migrating this PR from sparseml: https://github.com/neuralmagic/sparseml/pull/2306 

* add code for inferring compression format from model
* e2e fp8 example
* e2e tests

### CODE REVIEW

**Review by bfineran — APPROVED** [review]

File: tests/llmcompressor/transformers/compression/recipes/new_quant_fp8.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+quant_stage:
+  quant_modifiers:
+    GPTQModifier:
```
Comment (robertgshaw2-neuralmagic) [inline tests/llmcompressor/transformers/compression/recipes/new_quant_fp8.yaml:3]: Can we do this with just quantization modifier?

We shouldn't need GPTQ for this case

File: examples/quantization/llama7b_fp8_quantization.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+def preprocess(batch):
+    text = tokenizer.apply_chat_template(batch["messages"], tokenize=False)
+    tokenized = tokenizer(text, padding=True, truncation=True, max_length=2048)
+    return tokenized
+
+
+ds = load_dataset("mgoin/ultrachat_2k", split="train_sft")
+examples = ds.map(preprocess, remove_columns=ds.column_names)
+
+recipe = GPTQModifier(targets=["Linear"], scheme="FP8", ignore=["lm_head"])
```
Comment (robertgshaw2-neuralmagic) [inline examples/quantization/llama7b_fp8_quantization.py:25]: Is there a way to do fp8 without `GPTQModifier`?

Comment (robertgshaw2-neuralmagic) [reply examples/quantization/llama7b_fp8_quantization.py:25]: Resolved via slack

File: tests/llmcompressor/transformers/compression/recipes/new_quant_fp8.yaml (hunk 1 above)
Comment (robertgshaw2-neuralmagic) [reply tests/llmcompressor/transformers/compression/recipes/new_quant_fp8.yaml:3]: Can do as follow up

Comment (Satrat) [reply tests/llmcompressor/transformers/compression/recipes/new_quant_fp8.yaml]: Gotcha, I just updated the PR with this change (both in the example and the integration test). However the nice scheme/target UX is only for GPTQModifier so it made the example a bit messier. I'll move the scheme parsing to QM in a follow up PR

**Pull request merged**